### ISSUE TITLE
ESRI source fails quietly sometimes

### ISSUE THREAD

**Issue by NelsonMinar** [openaddresses/machine#40]
A few of our sources look like they run correctly but produce empty `out.csv` files. In each case it looks like the problem is the ESRI service is not returning any data. Maybe it's a bug in our ESRI scraping code or maybe it's just the service is broken. Either way we should catch this case and signal an error.

At the moment, I think the following sources show the problem:

ca-bc-surrey, us-ak-matanuska_susitna_borough, us-al-calhoun, us-mi-kent, us-mi-ottawa, us-mn-otter_tail, us-nv-las_vegas, us-pa-beaver, us-sc-berkeley, us-sc-lexington, us-sd, us-va-accomack, us-va-city_of_norton, us-va-essex, us-va-fluvanna, us-wi-fond_du_lac, us-wi-vernon, us-wy-laramie

**Comment by iandees** [comment]
The Python ESRI scrape code runs through OBJECTID's 500 at a time. Sometimes the source's OBJECTID first value is > 500, so the scrape returns zero results.

I should probably start off the scrape by doing an OBJECTID search from 0 to 1000000000 or something and then go from there.

**Comment by iandees** [comment]
I take that back. I'm assuming the column is called "OBJECTID" when in fact the oid column could be named something else. The name will be in the metadata for the layer so I'll query that first.

**Comment by NelsonMinar** [comment]
Hmm, I've tried a few sources after integrating your pull request #40 and still no joy. My repo is at f28c27f61a90d8 and us-sd us-al-calhoun us-sc-berkeley all still seem to find no features.

**Comment by NelsonMinar** [comment]
Oh btw here's how I'm testing stuff, after doing a `pip install`.
````
openaddr-process-one ~/src/oa/openaddresses-me/sources/us-al-calhoun.json /tmp/
wc -l /tmp/us-al-calhoun/out.csv
````

**Comment by NelsonMinar** [comment]
I've verified us-va-city_of_norton and us-wi-vernon now work. Also us-wy-albany still works, to name a random ESRI source that was working before #41 and #44 were merged. 

us-al-calhoun ran for about 10 minutes getting valid-seeming JSON and then barfed with a DownloadError, but that's likely to just be a failure of their web server (see below for details). We'll know more after the next bulk run. Closing the ticket, nice work @iandees!

````
  File "/home/nelson/src/oa/openaddresses-machine/openaddr/__init__.py", line 85, in cache
    downloaded_files = task.download(source_urls, workdir)
  File "/home/nelson/src/oa/openaddresses-machine/openaddr/cache.py", line 326, in download
    raise DownloadError("Problem querying ESRI dataset: %s", error['message'])
DownloadError: ('Problem querying ESRI dataset: %s', u'Error performing query operation')
````

**Comment by iandees** [comment]
The issue you're seeing there @NelsonMinar is fixed in #45.
